core/src/main/scala/org/broadinstitute/workbench/hamm/model.scala

```diff
@@ -1,17 +1,338 @@
+  }
+
+  case object F1Micro extends MachineType {
+    def asString = F1MICRO
+    def asDescriptionString: String = F1MICRO_DESCRIPTION_STRING
+    def asCPUresourceGroupString: String = F1MICRO
+    def asRAMresourceGroupString: String = F1MICRO
+    def asMetadataString: String = F1MICRO_METADATA_STRING
+  }
+
+  case object G1Small extends MachineType {
+    def asString = G1SMALL
+    def asDescriptionString: String = G1SMALL_DESCRIPTION_STRING
+    def asCPUresourceGroupString: String = G1SMALL
+    def asRAMresourceGroupString: String = G1SMALL
+    def asMetadataString: String = G1SMALL_METADATA_STRING
+  }
+}
+
+sealed trait Region {
+  def asString: String
+}
+object Region {
+  private final val GLOBAL = "global"
+  private final val US = "us"
+  private final val USCENTRAL1 = "us-central1"
+  private final val USEAST1 = "us-east1"
+  private final val USWEST4 = "us-east4"
+  private final val USWEST1 = "us-west1"
+  private final val USWEST2 = "us-west2"
+  private final val EUROPE =  "europe"
+  private final val EUROPEWEST1 = "europe-west1"
+  private final val EUROPEWEST2 = "europe-west2"
+  private final val EUROPEWEST3 = "europe-west3"
+  private final val EUROPEWEST4 = "europe-west4"
+  private final val EUROPENORTH1 = "europe-north1"
+  private final val NORTHAMERICANORTHEAST1 = "northamerica-northeast1"
+  private final val ASIA = "asia"
+  private final val ASIAEAST = "asia-east"
+  private final val ASIAEAST1 = "asia-east1"
+  private final val ASIAEAST2 = "asia-east2"
+  private final val ASIANORTHEAST = "asia-northeast"
+  private final val ASIASOUTHEAST = "asia-southeast"
+  private final val AUSTRALIASOUTHEAST1 = "australia-southeast1"
+  private final val AUSTRALIA = "australia"
+  private final val SOUTHAMERICAEAST1 = "southamerica-east1"
+  private final val ASIASOUTH1 = "asia-south1"
+
+  //ToDo: string might be in the format "us-central1-c" from Cromwell, handle those cases too
+  val stringToRegion = Map(
+      GLOBAL -> Global,
+      US -> Us,
+      USCENTRAL1 -> Uscentral1,
+      USEAST1 -> Useast1,
+      USWEST4 -> Uswest4,
+      USWEST1 -> Uswest1,
+      USWEST2 -> Uswest2,
+      EUROPE -> Europe,
+      EUROPEWEST1 -> Europewest1,
+      EUROPEWEST2 -> Europewest2,
+      EUROPEWEST3 -> Europewest3,
+      EUROPEWEST4 -> Europewest4,
+      EUROPENORTH1 -> Europenorth1,
+      NORTHAMERICANORTHEAST1 -> Northamericanortheast1,
+      ASIA -> Asia,
+      ASIAEAST -> Asiaeast,
+      ASIAEAST1 -> Asiaeast1,
+      ASIAEAST2 -> Asiaeast2,
+      ASIANORTHEAST -> Asianortheast,
+      ASIASOUTHEAST -> Asiasoutheast,
+      AUSTRALIASOUTHEAST1 -> Australiasoutheast1,
+      AUSTRALIA -> Australia,
+      SOUTHAMERICAEAST1 -> Southamericaeast1,
+      ASIASOUTH1 -> Asiasouth1 )
+
+  case object Global extends Region {
+    def asString = GLOBAL
+  }
+  case object Us extends Region {
+    def asString = US
+  }
+  case object Uscentral1 extends Region {
```

Comment:
Yeah :) It's mostly because Us (the case object) and US (the string val) - I could just change US to US_String and then capitalize all the Us's in the case objects